newIDE/app/src/ProjectManager/index.js

```diff
@@ -161,6 +163,26 @@ class Item extends React.Component<ItemProps, {||}> {
             label: 'Delete',
             click: () => this.props.onDelete(),
           },
+          {
```

Comment:
I see the idea with `typeAddedable`, but I think it would be simpler to pass a "addLabel" that is the label to be used for the menu item to add something. If addLabel is null/undefined/empty string, we'll hide it :)

Reply:
I understand, what you mean with `addLabel` ?
Or
I can simplify and delete visible options, i keep just one label named `Add a new one` linked to 
`this.props.onAdd()`

```
{
     label: 'Add a new one',
     click: () => this.props.onAdd(),
 },
```

Reply:
Consider the label as a prop, and only show the item if there is onAdd defined:

```
{
     label: this.props.addLabel,
     visible: !!this.props.onAdd, 
     click: () => this.props.onAdd(),
 },
```

Reply:
Ok `!!` convert value in boolean and invert boolean, very usefull thank you !

Reply:
Yes it's a way to convert ("cast") an object to a boolean (so that null/undefined => false, any object or array => true).